Automattic/newspack-plugin
# fix: show campaigns with pending and future status in UI

### All Submissions:






### Changes proposed in this Pull Request:



Current campaign UI only shows campaign posts that have `publish` and `draft` statuses. This would cause posts with the `pending` or `future` status to be dropped before being rendered in the UI.

This PR, along with https://github.com/Automattic/newspack-popups/pull/356, shows pending drafts under the "Draft" header and scheduled campaigns under the "Inactive" header, with a corresponding "Pending review" or "Scheduled" label for each.

<img width="275" alt="Screen Shot 2020-12-18 at 3 36 46 PM" src="https://user-images.githubusercontent.com/2230142/102667542-da3ed880-4146-11eb-918f-7dda900b5c79.png">

Closes https://github.com/Automattic/newspack-popups/issues/355.

### How to test the changes in this Pull Request:

1. On `master` for both this plugin and in Campaigns, create a new overlay or inline campaign and save it as a draft with the "Pending review" option enabled:

<img width="277" alt="Screen Shot 2020-12-18 at 3 37 56 PM" src="https://user-images.githubusercontent.com/2230142/102667602-05292c80-4147-11eb-91c5-89890e7331a7.png">

2. ~Go to **Newspack > Campaigns** and observe that the campaign is not rendered anywhere.~ Because of https://github.com/Automattic/newspack-popups/pull/399, you'll actually see these campaigns in the list, but they'll appear as drafts.
3. Check out this branch and run `npm run build`.
4. In **Newspack > Campaigns** under the Campaigns tab confirm that you now see the pending campaign with the "Draft" color with a "Pending review" label.
5. Edit the campaign again. Change the publish date to some date in the future and click "Schedule" button to schedule it for future publishing (the pending option is automatically disabled with the `future` and `publish` statuses).
6. Refresh the **Newspack > Campaigns** page. Confirm that the campaign still appears with the "Draft" color with a "Scheduled" label.
7. ~Turn on Test mode for the campaign and refresh the **Newspack > Campaigns** page. Confirm that the campaign still appears with the "Draft" color with a "Scheduled" label.~ There's no longer a test mode, so you can skip this step.

### Other information:
